homeassistant/components/onkyo/media_player.py

```diff
@@ -115,395 +196,350 @@
-        self._muted = bool(mute_raw[1] == "on")
-        if preset_raw and self._current_source.lower() == "radio":
-            self._attributes[ATTR_PRESET] = preset_raw[1]
-        elif ATTR_PRESET in self._attributes:
-            del self._attributes[ATTR_PRESET]
-        if self._supports_volume:
-            # AMP_VOL/MAX_RECEIVER_VOL*(MAX_VOL/100)
-            self._volume = (
-                volume_raw[1] / self._receiver_max_volume * (self._max_volume / 100)
-            )
-
-    @property
-    def supported_features(self):
-        """Return media player features that are supported."""
-        if self._supports_volume:
-            return SUPPORT_ONKYO
-        return SUPPORT_ONKYO_WO_VOLUME
-
-    def turn_off(self):
-        """Turn the media player off."""
-        self.command(f"zone{self._zone}.power=standby")
-
-    def set_volume_level(self, volume):
-        """
-        Set volume level, input is range 0..1.
-
-        However full volume on the amp is usually far too loud so allow the user to specify the upper range
-        with CONF_MAX_VOLUME.  we change as per max_volume set by user. This means that if max volume is 80 then full
-        volume in HA will give 80% volume on the receiver. Then we convert
-        that to the correct scale for the receiver.
-        """
-        # HA_VOL * (MAX VOL / 100) * MAX_RECEIVER_VOL
-        self.command(
-            f"zone{self._zone}.volume={int(volume * (self._max_volume / 100) * self._receiver_max_volume)}"
-        )
+            self._source = "_".join(values)
+
+    @callback
+    def _parse_sound_mode(self, sound_mode_raw):
+        values = _parse_onkyo_tuple(sound_mode_raw)
+
+        # If the selected sound mode is not available, N/A is returned
+        # so only update the sound mode when it is not N/A
+        if "N/A" not in values:
+            for sound_mode in values:
+                if sound_mode in SOUND_MODE_REVERSE_MAPPING:
+                    self._sound_mode = SOUND_MODE_REVERSE_MAPPING[sound_mode]
+                    break
+                self._sound_mode = "_".join(values)
+
+    @callback
+    def _parse_audio_inforamtion(self, audio_information_raw):
+        values = _parse_onkyo_tuple(audio_information_raw)
+        if "N/A" not in values:
+            info = {
+                "format": _tuple_get(values, 1),
+                "input_frequency": _tuple_get(values, 2),
+                "input_channels": _tuple_get(values, 3),
+                "listening_mode": _tuple_get(values, 4),
+                "output_channels": _tuple_get(values, 5),
+                "output_frequency": _tuple_get(values, 6),
+            }
+            self._attributes[ATTR_AUDIO_INFORMATION] = info
+        else:
+            self._attributes.pop(ATTR_AUDIO_INFORMATION, None)
+
+    @callback
+    def _parse_video_inforamtion(self, video_information_raw):
+        values = _parse_onkyo_tuple(video_information_raw)
```

Comment:
Ideally the pypi package is handling presenting this data to Home Assistant in a more consumable form that avoid using the magic numbers below.

Reply:
We removed the magic numbers by defining the audio and video information formats in a similar way as for the sound modes.